spec/requests/chat_channel_memberships_spec.rb

```diff
@@ -10,90 +10,301 @@
+    context "user is logged in" do
+      before do
+        chat_channel.add_users([second_user])
+      end
+
+      it "returns chat channel membership details" do
+        sign_in second_user
+        get "/chat_channel_memberships/find_by_chat_channel_id", params: {chat_channel_id: chat_channel.id}
+        expected_keys = %w(id status chat_channel_id last_opened_at channel_text
+                           channel_last_message_at channel_status channel_username
+                           channel_type channel_name channel_image
+                           channel_modified_slug channel_messages_count)
+        expect(JSON.parse(response.body).keys).to(match_array(expected_keys))
+      end
+    end
+
+    context "user is not logged in" do
+      it "renders not_found" do
+        expect do
+          get "/chat_channel_memberships/find_by_chat_channel_id", params: {}
+        end.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+  end
+
+  describe "GET /chat_channel_memberships/:id/edit" do
+    before do
+      chat_channel.add_users([second_user])
+    end
+
+    let(:chat_channel_membership) { chat_channel.chat_channel_memberships.where(user_id: second_user.id).first }
+
+    context "user is" do
+      context "not logged in" do
+        it "raise Pundit::NotAuthorizedError" do
+          expect do
+            get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+          end.to raise_error(Pundit::NotAuthorizedError)
+        end
+      end
+
+      context "user is logged in and channel id is wrong" do
+        it "raise ActiveRecord::RecordNotFound" do
+          sign_in second_user
+          expect do
+            get "/chat_channel_memberships/ERW/edit"
+          end.to raise_error(ActiveRecord::RecordNotFound)
+        end
+      end
+
+      context "channel member" do
+        it "allows user to view channel members" do
+          sign_in second_user
+          get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+          expect(response.body).to include("Members")
+          expect(response.body).to include("#{user.username}")
+          expect(response.body).to include("#{second_user.username}")
+          expect(response.body).not_to include("Pending Invitations")
+        end
+      end
+
+      context "channel moderator" do
+        it "allows user to view channel members" do
+          sign_in second_user
+          chat_channel_membership.update(role: "mod")
+          get "/chat_channel_memberships/#{chat_channel_membership.id}/edit"
+          expect(response.body).to include("Members")
+          expect(response.body).to include("#{user.username}")
+          expect(response.body).to include("#{second_user.username}")
+          expect(response.body).to include("Pending Invitations")
+          expect(response.body).to include("You are a channel mod")
+        end
+      end
+    end
+  end
+
+  describe "POST /chat_channel_memberships" do
+    context "user is" do
```

Comment:
I'm quite sure `rubocop-rspec` would flag these saying you need to use `when user is` or something like that :D